DFGM/dfgm_subsystem.py

```diff
@@ -0,0 +1,221 @@
+"""This python program represents a simulated version of the DFGM payload component for ExAlta3.
+
+The Digital Fluxgate Magnetometer (DFGM) is a payload on the satellite that automatically collects
+magnetic field data when it's on. Upon collecting this data, the DFGM board should place it into a
+packet along with some DFGM housekeeping data, then send the packet to the OBC for processing and
+saving.
+
+IMPORTANT NOTES:
+Data collection by the DFGM is done every second when it is on. Hence, there should be a data
+packet sent outward from the DFGM board every second as well.
+
+No commands are required to be sent to the DFMG board directly as its main purpose is to only
+record data and send it outward to another board (OBC) for processing/saving.
+
+Data sent by the DFGM board will be in a byte format; it's not readable if you print it out
+
+Usage: dfgm_subsystem.py non-default_port_num
+
+Copyright 2023 [Daniel Sacro]. Licensed under the Apache License, Version 2.0
+"""
+
+import sys
+import socket
+import time
+from struct import pack
+
+DEFAULT_HOST = '127.0.0.1'
+DEFAULT_PORT = 1802
+
+# Format/order of housekeeping data
+house_keeping_data = {
+    "Core Voltage": 5000, # HK 0 (mV)
+    "Sensor Temperature": 25, # HK 1 (deg C)
+    "Reference Temperature": 25, # HK 2 (deg C)
+    "Board Temperature": 25, # HK 3 (deg C)
+    "Positive Rail Voltage": 5000, # HK 4 (mV)
+    "Input Voltage": 5000, # HK 5 (mV)
+    "Reference Voltage": 5000, # HK 6 (mV)
+    "Input Current": 1000, # HK 7 (mA)
+    "Reserved 1": 0, # HK 8 (Unused)
+    "Reserved 2": 0, # HK 9 (Unused)
+    "Reserved 3": 0, # HK 10 (Unused)
+    "Reserved 4": 0, # HK 11 (Unused)
+}
+
+# Format of a raw magnetic field data sample to be processed by the OBC
+magnetic_field_tuple = {
+    "x_DAC": 1, 
+    "x_ADC": 1,
+    "y_DAC": 2,
+    "y_ADC": 2,
+    "z_DAC": 3,
+    "z_ADC": 3
+}
+
+# There are 100 samples in each packet from the DFGM
+magnetic_field_data = [magnetic_field_tuple] * 100
```

Comment:
Again I think we should avoid the use of 'magic numbers' here and define the 100 somewhere, maybe as a const. Assuming there are always 100 samples in each packet.